openmdao/docs/features/building_blocks/components/spline_comp.rst

```diff
@@ -0,0 +1,152 @@
+between these points. We call these new fixed locations at which to interpolate: `x`. When we
+instantiate a `SplineComp`, we specify these `x_cp` and `x` locations as numpy arrays and pass
+them in as constructor keyword arguments. (Figure 1). Next we'll add our `y_cp` data in by
+calling the `add_spline` method and passing the `y_cp` values in as the keyword argument `y_cp_val` (Figure 2).
+`SplineComp` computes and outputs the `y_interp` values (Figure 3).
+
+.. image:: images/figure_1.png
+  :width: 900
+
+.. image:: images/figure_2.png
+  :width: 900
+
+.. image:: images/figure_3.png
+  :width: 900
+
+.. image:: images/figure_4.png
+  :width: 900
+
+.. embed-code::
+    openmdao.components.tests.test_spline_comp.SplineCompFeatureTestCase.test_basic_example
+    :layout: code
+
+
+SplineComp Multiple Splines
+---------------------------
+
+`SplineComp` supports multiple splines on a fixed `x_interp` grid. Below is an example of how a user can
+setup two splines on a fixed grid. To do this the user needs to pass in names to give to the component
+input and output. The initial values for `y_cp` can also be specified here.
+
+.. embed-code::
+    openmdao.components.tests.test_spline_comp.SplineCompFeatureTestCase.test_multi_splines
+    :layout: code
+
+
+Specifying Options for 'akima'
+------------------------------
+
+When you are using the 'akima' method, there are two akima-specific options that can be passed in to the
+`SplineComp` constructor.  The 'delta_x' option is used to define the radius of the smoothing interval
+that is used in the absolute values functions in the akima calculation in order to make their
+derivatives continuous.  This is set to zero by default, which effectively turns off the smoothing.
+The 'eps' option is used to define the value that triggers a division-by-zero
+safeguard; its default value is 1e-30.
+
+
+.. embed-code::
+    openmdao.components.tests.test_spline_comp.SplineCompFeatureTestCase.test_akima_options
+    :layout: code
+
+
+Specifying Options for 'bsplines'
+---------------------------------
+
+When you use the 'bsplines' method, you can specify the bspline order by defining 'order' in an
+otherwise empty dictionary and passing it in as 'interp_options'.
+
+In addition, when using 'bsplines', you cannot specify the 'x_cp' locations because the bspline
+formulation differs from other polynomial interpolants. When using bsplines, you should instead
+specify the number of control points using the 'num_cp' argument.
+
+.. embed-code::
+    openmdao.components.tests.test_spline_comp.SplineCompFeatureTestCase.test_bspline_options
+    :layout: code
+
+
+SplineComp Interpolation Distribution
+-------------------------------------
+
+We have included three different distribution functions for users to replicate functionality that used to
+be built in to the individual akima and bsplines components. The `cell_centered` function takes the number
```

Comment:
`built-in`  

(https://www.merriam-webster.com/dictionary/built-in, had to check ;))

Reply:
I have completely rewritten this paragraph, so built-in isn't in there anymore.

Reply:
Disregard -- "built in" is still there, and is now "built-in".